### PR TITLE
Shadow for new Http Client

### PR DESCRIPTION
**Pull request by gitaumoses4**
## What does this PR do?

Updates the shadows for HttpRequest to use the new HttpClient.

The update was done [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1161)

## Related discussion
https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1270#issuecomment-756494761

### CODE REVIEW

**Comment by shahzaibj** [comment]
@gitaumoses4 Tests are failing...I believe this is because submodule is out of date. You would need to update the common submodule to point to the latest code on `dev` so that the HttpClient code from common gets used in these tests.

**Review by shahzaibj — CHANGES_REQUESTED** [review]
Tests are failing. Need to update common submodule.

**Review by shahzaibj — CHANGES_REQUESTED** [review]
The tests still seem to be failing. Were they passing on you local machine? 

The reason for these failures now I believe to be an issue with the shadow itself. Since what you are shadowing now is no longer a static method but rather an instance method, I believe you probably need to put the `@Implementation` annotation on the shadowed methods. See this for reference: http://robolectric.org/extending/#:~:text=Shadow%20objects%20implement%20methods%20that,the%20Android%20object%20is%20invoked.

**Comment by gitaumoses4** [comment]
Hi @shahzaibj , The tests are now passing locally. Only the `SingleAccountOverloadsMockedTest` is failing. Still trying to figure out why.

File: msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java [hunk 1]
```diff
@@ -71,21 +72,23 @@
...
 
     @Before
     public void setup() {
         super.setup();
         TestUtils.clearCache(SingleAccountPublicClientApplication.SINGLE_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES);
         mSingleAccountPCA = (SingleAccountPublicClientApplication) mApplication;
-        MockHttpResponse.setHttpResponse(MockServerResponse.getMockTokenSuccessResponse());
+
+        MockHttpResponse.setHttpResponse(MockServerResponse.getMockTokenSuccessResponse(), HttpClient.HttpMethod.POST);
+        MockHttpResponse.setHttpResponse(MockServerResponse.getMockCloudDiscoveryResponse(), HttpClient.HttpMethod.GET);
```
Comment (shahzaibj) [inline msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java:91]: should we specific about url here?

File: msal/src/test/java/com/microsoft/identity/client/e2e/shadows/ShadowHttpClient.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+@Implements(AbstractHttpClient.class)
+public class ShadowHttpClient {
```
Comment (shahzaibj) [inline msal/src/test/java/com/microsoft/identity/client/e2e/shadows/ShadowHttpClient.java:43]: add javadoc

File: msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java (hunk 1 above)
Comment (gitaumoses4) [reply msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java]: Yes, I have added the regex.

File: msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java [hunk 3]
```diff
@@ -71,21 +72,23 @@
 @Config(shadows = {
         ShadowStorageHelper.class,
         ShadowAuthorityForMockHttpResponse.class,
-        ShadowHttpRequestForMockedTest.class,
         ShadowMsalUtils.class,
-        ShadowOpenIdProviderConfigurationClient.class
+        ShadowHttpClient.class, ShadowOpenIdProviderConfigurationClient.class
```
Comment (shahzaibj) [inline msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountOverloadsMockedTest.java:76]:
```suggestion
        ShadowHttpClient.class, 
        ShadowOpenIdProviderConfigurationClient.class
```

**Review by shahzaibj — APPROVED** [review]

**Review by tanmaymanolkar1 — APPROVED** [review]

**Pull request merged**